addon/scheduler.ts

```diff
@@ -58,8 +59,15 @@ export function setupRouter(router: Router) {
   }
 
   (router as any)[APP_SCHEDULER_HAS_SETUP] = true;
-  router.on('willTransition', beginTransition);
-  router.on('didTransition', endTransition);
+  let owner = getOwner(router);
+  let service = owner.lookup('service:router');
+  if (service) {
```

Comment:
BTW, the router server has been available since around 2.17 but these events have only been around since 3.6.0.

https://github.com/ember-app-scheduler/ember-app-scheduler/pull/189 uses ember-compatibility-helpers to avoid extra code being shipped depending on version (I know I know, its only a couple lines..)